test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodConnectivityReportDataWithSenderAndReceiverSource.cs

```diff
@@ -50,6 +50,24 @@ class DirectMethodConnectivityReportDataWithSenderAndReceiverSource
                     10, 6, 1, 0, 0, 0, 0, 0, 0, true
                 },
                 new object[]
+                {
+                    // NetworkOffSuccess test with FailedDependency exception
+                    Enumerable.Range(1, 7).Select(v => (ulong)v),
+                    new[] { 1UL, 2UL, 4UL, 5UL, 6UL, 7UL },
+                    new List<HttpStatusCode> { HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.FailedDependency, HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.OK, HttpStatusCode.OK },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 16, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 1, 0, 0, 0, 0, 0, 0, true
+                },
```

Comment:
This test relates to the new case I added where I treat failed dependency as passing when offline.